src/lib.rs

```diff
@@ -62,6 +63,7 @@ pub struct Peer {
     outbound_sender: Sender<MessageBeanOut>,
     ktable: RwLock<Tree<PeerInfo>>,
     header: Header,
+    blocklist: RwLock<HashSet<SocketAddr>>,
```

Comment:
As I suppose we need to block remote addresses by IP (for any port), how this is handled if the `cmp` method compares the port as well?

```rust
impl Ord for SocketAddrV4 {
    fn cmp(&self, other: &SocketAddrV4) -> Ordering {
        self.ip().cmp(other.ip()).then(self.port().cmp(&other.port()))
    }
}
```

Reply:
I get your point, and maybe it's a good thing to block the whole ip.
However, since the current ID is derived by `ip`+`port` and the distance from that ID is the bucket height, we need to keep the port otherwise we need to full scan the whole routing table in order to remove that peer.

Imo this feature should be interpreted as a temp-patch until the user read the logged warning somewhere, and block the IP properly.

Anyway, even with this soft blocklist maybe worth to block the whole IP, but I would like to discuss it in a separate issue/discussion